crates/astria-conductor/src/conductor/inner.rs

```diff
@@ -54,24 +48,18 @@ impl std::fmt::Display for RestartOrShutdown {
     }
 }
 
-enum ExitReason {
-    ShutdownSignal,
-    TaskFailed {
-        name: &'static str,
-        error: eyre::Report,
-    },
-}
+struct ShutdownSignalReceived;
 
 pin_project! {
     /// A handle returned by [`ConductorInner::spawn`].
     pub(super) struct InnerHandle {
```

Comment:
I think we can probably get rid of this type altogether?  The shutdown token seems to be unused, and likewise the nested `task` member is never `None`.

We could just use an alias like:
```rust
pub(super) type InnerHandle = JoinHandle<eyre::Result<RestartOrShutdown>>;
```

Reply:
Yeah, good observation. I have removed `InnerHandle` entirely and renamed `ConductorInner` to just `Inner`.

(not going the type alias route though because it's just used in one place)